shoreditch.php

```diff
@@ -199,5 +199,13 @@ function shoreditch_civicrm_navigationMenu(&$menu) {
  *   TRUE if Shoreditch is the active theme.
  */
 function _shoreditch_isActive() {
+  // The Job civicrm_api3_job_cleanup can clear the DB cache, causing the
+  // CiviCRM menu to be empty on the first load after the command ran.
+  // Without the menu, the getActiveThemeKey is unable to detect the current
+  // theme. This condition ensures the menu is filled.
+  if (!CRM_Core_DAO::checkTableHasData(CRM_Core_DAO_Menu::getTableName())) {
+    CRM_Core_Menu::store(FALSE);
+  }
```

Comment:
Glad about figuring out the root case of this issue. Although, the code perfectly solves it, I am not in the fever of doing it here at theme level because it affects all CiviCRM themes hence it is a CiviCRM issue.
Maybe we should move the code to [DrupalBase.php#L662](https://github.com/civicrm/civicrm-core/blob/18c0020ce567f61bf0acaa087827c70669f2662e/CRM/Utils/System/DrupalBase.php#L662) or better to build the menu as part of the `civicrm_api3_job_cleanup`  like in [CRM/Upgrade/Form.php#L783](https://github.com/civicrm/civicrm-core/pull/17126/commits/3af5c3fe5f91800f582527da31b089c3bfbedb7d) - see the line  `CRM_Core_Invoke::rebuildMenuAndCaches(FALSE, TRUE);`